six/modules/c++/six.sicd/include/six/sicd/CropUtils.h

```diff
@@ -116,6 +116,21 @@ void cropSICD(six::NITFReadControl& reader,
               const std::vector<scene::LatLonAlt>& corners,
               const std::string& outPathname,
               bool trimCornersIfNeeded = true);
+
+/*
+ * Given a six::Sicd::ComplexData object and a cropping region,
+ * update the meta data in the same way as the functions above.
+ * \param complexData the input complexData object
+ * \param aoiOffset Upper left corner of AOI
+ * \param aoiDims Size of AOI
+ *
+ * \returns A cloned complex data object where the 
+ * meta data has been updated to refelect croped aoi.
+ */
+six::sicd::ComplexData* const
```

Comment:
Return by `auto_ptr` rather than raw pointer.  Swig should be able to deal with this.